compiler/rustc_next_trait_solver/src/solve/eval_ctxt/mod.rs

```diff
@@ -174,9 +175,10 @@ where
         &self,
         goal: Goal<I, I::Predicate>,
         generate_proof_tree: GenerateProofTree,
+        span: I::Span,
     ) -> (Result<(HasChanged, Certainty), NoSolution>, Option<inspect::GoalEvaluation<I>>) {
         EvalCtxt::enter_root(self, self.cx().recursion_limit(), generate_proof_tree, |ecx| {
-            ecx.evaluate_goal(GoalEvaluationKind::Root, GoalSource::Misc, goal)
+            ecx.evaluate_goal(GoalEvaluationKind::Root, GoalSource::Misc, goal, span)
```

Comment:
vibeck: should we move the span to a field of the evalctxt instead?

Reply:
i guess it could be? origin_span or something.